cfme/utils/appliance/__init__.py

```diff
@@ -2135,45 +2144,71 @@ def check_domain_enabled(self, domain):
-                                 remote_file=ldap_yaml.cert_filepath)
+        self.ssh_client.put_file(local_file=conf_path.join(cert_filename).strpath,
+                                 remote_file=cert_filepath)
         # configure ldap and sssd with conf file content from yaml
-        assert self.ssh_client.run_command('echo "{}"  > /etc/openldap/ldap.conf'
-                                           .format(ldap_yaml.ldap_conf))
-        assert self.ssh_client.run_command('echo "{}" > /etc/sssd/sssd.conf'
-                                           .format(ldap_yaml.sssd_conf))
-        assert self.ssh_client.run_command('chown -R root:root /etc/sssd/sssd.conf')
-        assert self.ssh_client.run_command('chmod 600 /etc/sssd/sssd.conf')
-        # copy miq/cfme template files for httpd ext auth config
-        template_dir = '/opt/rh/cfme-appliance/TEMPLATE'
-        if self.version == 'master':
-            template_dir = '/var/www/miq/system/TEMPLATE'
-        manageiq_ext_auth = '/etc/httpd/conf.d/manageiq-external-auth.conf'
         assert self.ssh_client.run_command(
-            'cp {template_dir}/etc/pam.d/httpd-auth /etc/pam.d/httpd-auth'
-            .format(template_dir=template_dir))
+            'echo "{s}" > {c}'.format(s=ldap_conf, c=self.CONF_FILES['openldap'])
+        )
         assert self.ssh_client.run_command(
-            'cp {template_dir}/etc/httpd/conf.d/manageiq-remote-user.conf /etc/httpd/conf.d/'
-            .format(template_dir=template_dir))
+            'echo "{s}" > {c}'.format(s=sssd_conf, c=self.CONF_FILES['sssd'])
+        )
+        assert self.ssh_client.run_command('chown -R root:root {}').format(self.CONF_FILES['sssd'])
+        assert self.ssh_client.run_command('chmod 600 {}').format(self.CONF_FILES['sssd'])
+        # copy miq/cfme template files for httpd ext auth config
+        template_dir = self.CONF_FILES.get(
+            'downstream_templates' if self.is_downstream else 'upstream_templates'
+        )
+        # pam httpd-auth and httpd remote-user.conf
+        for file in [self.CONF_FILES['pam_httpd_auth'], self.CONF_FILES['httpd_remote_user']]:
+            assert self.ssh_client.run_command('cp {t}{c} {c}'.format(t=template_dir, c=file))
+
+        # https external-auth conf, template has extra '.erb' suffix
         assert self.ssh_client.run_command(
-            'cp {template_dir}/etc/httpd/conf.d/manageiq-external-auth.conf.erb {manageiq_ext_auth}'
-            .format(template_dir=template_dir,
-                    manageiq_ext_auth=manageiq_ext_auth))
-        self.ssh_client.run_command('setenforce 0 && '
-                                    'systemctl restart sssd && '
-                                    'systemctl restart httpd')
+            'cp {t}{c}.erb {c}'
+            .format(t=template_dir, c=self.CONF_FILES['httpd_ext_auth'])
+        )
+        assert self.ssh_client.run_command('setenforce 0')
```

Comment:
I would rather check selinux state instead. I doubt it will work in containers.

Reply:
This comes straight from the MIQ documentation on LDAP configuration:

http://manageiq.org/docs/reference/latest/auth/ldap

Reply:
okay, lets leave it until we run into some issue